src/EditorFeatures/CSharpTest/BlockCommentCompletion/BlockCommentCompletionTests.cs

```diff
@@ -0,0 +1,346 @@
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine1()
+        {
+            var code = @"
+    /*
+     *$$*/
+";
+            var expected = @"
+    /*
+     *
+     $$*/
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine2()
+        {
+            var code = @"
+    /*
+     *$$ */
+";
+            var expected = @"
+    /*
+     *
+     *$$ */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine3()
+        {
+            var code = @"
+    /*
+     * $$ 1.
+     */
+";
+            var expected = @"
+    /*
+     * 
+     * $$ 1.
+     */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine4()
+        {
+            var code = @"
+    /*
+     *  1.$$
+     */
+";
+            var expected = @"
+    /*
+     *  1.
+     *  $$
+     */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine5()
+        {
+            var code = @"
+    /*
+     *   1.
+     *   $$
+     */
+";
+            var expected = @"
+    /*
+     *   1.
+     *   
+     *   $$
+     */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine6()
+        {
+            var code = @"
+    /*
+  $$   *
+     */
+";
+            var expected = @"
+    /*
+  
+     * $$   *
+     */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine7()
+        {
+            var code = @"
+    /*
+     *************$$
+     */
+";
+            var expected = @"
+    /*
+     *************
+     * $$
+     */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnEndLine0()
+        {
+            var code = @"
+    /*
+     *$$/
+";
+            var expected = @"
+    /*
+     *
+     *$$/
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnEndLine1()
+        {
+            var code = @"
+    /*
+  $$   */
+";
+            var expected = @"
+    /*
+  
+     * $$   */
```

Comment:
Not sure i agree with this one.   if the user isn't using * to preface the lines, then i don't think we should start adding.  essentially:

1. if your previously line starts with a ```*``` (or is ```/*```), then you can start with ```*```.  Otherwise you shoudln't.